cmd/update.go

```diff
@@ -0,0 +1,197 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	"github.com/anchore/grype/internal/log"
+	"github.com/anchore/grype/internal/version"
+)
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "updates grype",
+	RunE:  runUpdate,
+}
+
+func init() {
+	rootCmd.AddCommand(updateCmd)
+}
+
+func runUpdate(_ *cobra.Command, _ []string) error {
+	fmt.Println("Checking for a newer version...")
+	newerVersionAvailable, desiredVersion, err := checkLatestVersion()
+	if err != nil {
+		panic(err)
+	}
+	if newerVersionAvailable {
+		compressed := downloadCompressed(desiredVersion)
+		filePath := extractBinary(compressed)
+		replaceBinary(filePath)
+	} else {
+		fmt.Println("You are already running the newest version available")
+	}
+	return nil
+}
+
+func extractBinary(fileName string) string {
+	r, err := os.Open(fileName)
+	if err != nil {
+		defer os.Remove(fileName)
+		panic(err)
+	}
+	tempDir, err := os.MkdirTemp("", "e_grype")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Extracting grype...")
+	log.Infof("Extracting grype")
+	log.Debugf("Extract destination is %s", tempDir)
+	err = ExtractTarGz(r, tempDir)
+	if err != nil {
+		defer os.RemoveAll(tempDir)
+		panic(err)
+	}
+	updatedGrypePath := fmt.Sprintf("%s/grype", tempDir)
+	err = os.Chmod(updatedGrypePath, 0777)
+	if err != nil {
+		panic(err)
+	}
+	return updatedGrypePath
+}
+
+func replaceBinary(fileName string) {
+	fmt.Println("Applying update...")
```

Comment:
nit: try not to mix and match the `log/fmt` packages here for grype

consider using the TUI that's been established and used through the other commands